### PR TITLE
feat(docs): update Pill tabs to use new layout

### PR DESCRIPTION
**Pull request by MariaJose**
## What?
Update pill tabs to use new layout.

## Screenshots
Screen Shot 2022-02-17 at 4 18 28 PM

### CODE REVIEW

File: packages/docs/PropTables/PillTabsPropTable.tsx [hunk 1]
```diff
@@ -11,12 +11,12 @@ const pillTabsPropTable: Prop[] = [
   {
     description: (
       <>
-        See <NextLink href="#pill-tabs-items-prop-table">below</NextLink> for usage.
+        See <NextLink href="/pill-tabs?props=pill-tab-item#props">here</NextLink> for usage.
```
Comment (chanceaclark) [inline packages/docs/PropTables/PillTabsPropTable.tsx:14]: We did something a little differently for the other pages. See the [TablePropTable](https://github.com/bigcommerce/big-design/blob/master/packages/docs/PropTables/TablePropTable.tsx#L11) as an example.

**Review by chanceaclark — COMMENTED** [review]
Looks great, beside one little comment!

File: packages/docs/PropTables/PillTabsPropTable.tsx (hunk 1 above)
Comment (MariaJose) [reply packages/docs/PropTables/PillTabsPropTable.tsx:14]: sounds good, I'll take a look. Ty!

File: packages/docs/pages/pill-tabs.tsx [hunk 2]
```diff
@@ -1,110 +1,134 @@
...
+        />
+      </Panel>
+
+      <Panel header="Do's and Don'ts" headerId="guidelines">
+        <GuidelinesTable
+          recommended={[
+            <>Use on pages that have a large amount of content.</>,
+            <>Be concise on the navigation labels, ideally one or two words rather than a phrase.</>,
+            <>
+              Default page view should have no <Code primary>PillTabs</Code> selected.
```
Comment (MariaJose) [inline packages/docs/pages/pill-tabs.tsx:120]: Not sure if we should update this sentence (line 120) like this:
`When filtering a collection of items, the default page view should have no <Code primary>PillTabs</Code> selected`

**Comment by MariaJose** [comment]
♻️

**Review by jorgemoya — APPROVED** [review]

File: packages/docs/pages/pill-tabs.tsx (hunk 2 above)
Comment (jorgemoya) [reply packages/docs/pages/pill-tabs.tsx:120]: To me this change reads as if we have already clicked a pill tab (thus filtered).

Comment (MariaJose) [reply packages/docs/pages/pill-tabs.tsx:120]: makes sense, Initially I thought that pill tabs were used as tabs, and not only filters. Maybe we could leave it just as it is.

**Review by chanceaclark — APPROVED** [review]

**Pull request merged**